desktop-widgets/printerwebengine.cpp

```diff
@@ -46,8 +46,10 @@ void Printer::onLoadFinished()
 					  img.style.width = \"100%\"; \
 					  profile.appendChild(img); \
 					} \
-				", [this](const QVariant &v) { emit profilesInserted(); });
+				", [](const QVariant &v) {;});
 
+	} else {
+		emit profilesInserted();
```

Comment:
I have not looked at the code, but this function looks ominous.

You say that after the call above the profiles are not rendered. Yet, you set `profilesMissing` to false. Seems like a wrongly named flag?

Also, this seems to imply that `onLoadFinished()` is called multiple times - please add a code-comment detailing the control flow. The casual reader will be confused.

From a quick glance I only see that there are multiple(!) `connect()` calls, but no corresponding `disconnect()`s. What is going on there? Needs comments.

I don't think I will find time to look at this today, but for now, out of caution, I would NAK this. Needs more explanation of what is going on.

Reply:
Let's see first if this solves the problems @dirkhh encountered on Linux. This is a call back that is run whenever the web view finished loading a page. The first time (then profilesMissing is still true) it is called once the print.html file is read from disk. This time, we need to invoke the JS that adds the <img> tags to the page. This modified page is then rendered once more, now with the profile images and after that finished, the call back is called once more, this time with profilesMissing==false. This is then time to ship out to the printer. If this is really the solution, I will add comments explaining this.

Reply:
What about the two `connect()` statements? Seems like a recipe for disaster: wouldn't the callback be called twice, resulting in timing issues?

And why on earth do we need JavaScript to render images? I stopped wondering about Qt-APIs, but that is ... very strange.

Moreover, I wonder why does executing a JavaScript on a page result in an `onLoadFinished()` signal? Is that per image or at the end of the script?